core/src/client/client.rs

```diff
@@ -79,6 +81,8 @@ pub struct Client {
     genesis_accounts: Vec<Address>,
 
     importer: Importer,
+
+    reseal_timer: RwLock<Option<TimerApi>>,
```

Comment:
I think it as a code smell to have ```Option<>``` and intermediate mutation methods(```register_reseal_timer```, ```register_timer```) everywhere. In this case, I think Timer should have a remedy to this situation. And there's also hidden Arc reference cycle between Timer <-> Client.

Reply:
It's inevitable by the code now. The network code is same.

Reply:
Okay, I understand. I think I need to fix the Timer to make it handles intermediate mutation.

Reply:
Currently, the flow is:
1. Make `TimerLoop`
2. Make struct with `TimeoutHandler`
3. Register the struct to `TimerLoop`, and get `TimerApi`
4. Register `TimerApi` to the struct.

The problem causes between 2 and 3.